### ISSUE TITLE
Stats Expandable View Does Not Show Collapse Animation

### ISSUE THREAD

**Issue by theck13** [wordpress-mobile/WordPress-Android#9171]
### Expected
The chevron icon in an expandable list row animates when the view is expandable and collapsed.

### Observed
The chevron icon in an expandable list row animates when the view is expandable, but not collapsed.  See the animation below for illustration.

![stats_animation_collapse](https://user-images.githubusercontent.com/3827611/52186531-1bdd6900-27e5-11e9-93cc-03c4a0bd9184.gif)

### Reproduced
0. Select site with at least one post.
1. Go to ***Sites*** tab.
2. Tap ***Stats*** item.
3. Tap ***Years*** tab.
4. Tap author in ***Authors*** section.
5. Notice chevron animates when view is expanded.
6. Tap author in ***Authors*** section from Step 4.
6. Notice chevron does not animate when view is collapsed.

##### Tested
Google Pixel 3 on Android 9.0 with alpha-150

**Comment by planarvoid** [comment]
thanks for reporting this @theck13 ! This is something we're aware of and we tried to find a good solution and fell back to this. This is my comment in the code(`org.wordpress.android.ui.stats.refresh.lists.sections.BlockListAdapter`, please let me know if you know of a solution to this issue! 
```
// We're using a nested recycler view here. When we try to update a block with new data, there are 2 animations
// happening at the same time. The outer recycler view is animating the block height change and the inner
// recycler view is animating items being added or removed. The problem is that only the inner animation
// is actually animating, the block size happens immediately. This looks OK for adding new items to the list
// or when the size of the list doesn't change. In that case the block height increases immediately and the
// inner list changes are animated.
// However, when we want to remove items, the block height decreases immediately so the bottom items disappear
// and then the animation happens and they slowly slide back into view. This doesn't look good. We couldn't
// find a solution that handles this case well so we're falling back to notifying the whole list instead of
// animating changes when we decrease the number of items. When we do that, the outer animation happens and
// the items are replaced immediately.
```

**Comment by theck13** [comment]
Did you try doing the animations in sequence?  That's what we did for the Quick Start v2 expand/collapse animations.  First, we animate the chevron icon rotating.  Second, we add/remove items from the adapter once the first animation completes.  We set an `AnimatorListener` on the first animation so we know when to start the second animation in `onAnimationEnd`.  See the animation below for illustration.

![qsv2](https://user-images.githubusercontent.com/3827611/52290316-0a06dd80-292d-11e9-86c3-265bd2061926.gif)

Notice the ripple effect on the Quick Start v2 expandable view when it expands and collapses.  There is a ripple effect on the Stats expandable view when it expands, but not when it's collapsed.  I'm guessing the Stats expandable view is being updated at the same time the collapse animation starts, which is why the ripple isn't shown either.  Sequencing the animations should fix that as well.
